packages/block-editor/src/components/observe-typing/index.js

```diff
@@ -18,7 +18,7 @@ import {
 	BACKSPACE,
 	ESCAPE,
 } from '@wordpress/keycodes';
-import { withSafeTimeout, compose } from '@wordpress/compose';
+import { withSafeTimeout } from '@wordpress/compose';
```

Comment:
I think you could avoid using `withSafeTimeout` and just use `setTimeout` in an effect. See https://github.com/WordPress/gutenberg/pull/19109#issuecomment-565397387

Reply:
It’s a bit more complicated if it’s called in a event. That’s why I created that PR, but I think it’s such a rare thing to do that we shouldn’t create a helper for it.

Reply:
It could be done here by storing the timeout id as a ref

Reply:
> It could be done here by storing the timeout id as a ref

Yeah, though, this is true of any component even prior to hooks, prior to `withSafeTimeout`. The goal of the higher-order component is more a convenience to make sure the developer does it correctly. I think with the ESLint rule added in #16292, we can have a little more confidence here that it's difficult to do it _wrong_, so I guess there's not as much value/need in having the higher-order component.

It's a little less expressive without the higher-order component, or at least harder to follow only in that the alternative is not nearly as concise / colocated as it was previously:

```js
let stopTypingTimeoutId;

// ... some code

useEffect( () => () => clearTimeout( stopTypingTimeoutId ) );

// ... some code

function stopTypingOnNonTextField() {
	stopTypingTimeoutId = setTimeout( /* ... */ );
}
```

Maybe this is a good thing, in that it discourages a developer from reaching to the (often misguided) conclusion that `setTimeout` is the valid solution. 🤷‍♂

Reply:
I think for now I'll keep `withSafeTimeout`. We could consider to address separately, though I think it's the sort of thing where we should sooner concern ourselves with avoiding the timer altogether, than to spend the effort to optimize its usage.